File: client/fix_styles.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

walkDir('/home/sneha/SpendSense/client/src', function(filePath) {
  if (filePath.endsWith('.css')) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Replace hardcoded primary colors
    content = content.replace(/#2563EB/gi, 'var(--primary)');
    content = content.replace(/#1D4ED8/gi, 'var(--primary-hover)');
    
    // Replace all pixel-based border radiuses with the variable
    content = content.replace(/border-radius:\s*[1-9][0-9]*px;/g, 'border-radius: var(--radius-lg);');
    
    // Replace box-shadows (ignoring those that are already variables or none)
    content = content.replace(/box-shadow:\s*0[^;var]+;/g, 'box-shadow: var(--shadow);');

    fs.writeFileSync(filePath, content);
  }
});
console.log("CSS files updated successfully.");

File: client/fix_styles.py
import os
import glob
import re

def process_css_files():
    css_files = glob.glob('/home/sneha/SpendSense/client/src/**/*.css', recursive=True)
    
    for file in css_files:
        with open(file, 'r') as f:
            content = f.read()
            
        # Colors
        content = re.sub(r'#2563EB', 'var(--primary)', content, flags=re.IGNORECASE)
        content = re.sub(r'#1D4ED8', 'var(--primary-hover)', content, flags=re.IGNORECASE)
        
        # Border radiuses
        content = re.sub(r'border-radius:\s*[1-9][0-9]*px;', 'border-radius: var(--radius-lg);', content)
        
        # Shadows (simple heuristic)
        content = re.sub(r'box-shadow:\s*0[^;]+;', 'box-shadow: var(--shadow);', content)
        
        with open(file, 'w') as f:
            f.write(content)

if __name__ == "__main__":
    process_css_files()
